stock_landed_cost_company_percentage/models/stock_picking.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    def prepare_landed_cost_lines(self, company, amount):
+        cost_lines = []
+        for line in company.landed_cost_company_line:
+            values = {
+                'product_id': line.product_id.id,
+                'split_method': line.product_id.split_method,
+                'account_id': line.product_id.property_account_expense_id.id or
+                              line.product_id.categ_id.property_account_expense_categ_id.id, # noqa
+                'price_unit': float((amount or 0.0) * line.percentage) / 100.0,
+                'name': line.product_id.name,
+            }
+            cost_lines.append((0, 0, values))
+        return cost_lines
+
+    def prepare_landed_cost(self, amount):
+        return {
+            'picking_ids': [(6, 0, [self.id])],
+            'account_journal_id': self.company_id.journal_id.id,
+            'cost_lines': self.prepare_landed_cost_lines(
+                self.company_id, amount)
+        }
+
+    @api.multi
+    def action_done(self):
+        res = super(StockPicking, self).action_done()
+        for pick in self:
```

Comment:
Could you extract this code in another method and call it from `action_done`? Including it directly in `action_done` can be a problem for the extensibility.

Reply:
Corresponded. Thanks.